discovery-provider/es-indexer/etl/trackEtl.ts

```diff
@@ -0,0 +1,168 @@
+        // reposts
+        reposted_by: { type: 'keyword' },
+        repost_count: { type: 'integer' },
+
+        artist: {
+          properties: {
+            handle: { type: 'keyword' },
+            location: { type: 'keyword' },
+            name: { type: 'text' }, // should it be keyword with a `searchable` treatment?
+            follower_count: { type: 'integer' },
+          },
+        },
+
+        stem_of: {
+          properties: {
+            category: { type: 'keyword' },
+            parent_track_id: { type: 'keyword' },
+          },
+        },
+
+        'remix_of.tracks.parent_track_id': { type: 'keyword' },
+      },
+    },
+  },
+
+  sql2: (checkpoint: BlocknumberCheckpoint) => {
+    let sql = `
+    -- etl tracks
+    select 
+      tracks.*,
+
+      json_build_object(
+        'handle', users.handle,
+        'name', users.name,
+        'location', users.location,
+        'follower_count', follower_count
+      ) as artist,
+
+      array(
+        select slug 
+        from track_routes r
+        where
+          r.track_id = tracks.track_id
+        order by is_current
+      ) as routes,
+
+      array(
+        select user_id 
+        from reposts
+        where
+          is_current = true
+          and is_delete = false
+          and repost_type = 'track' 
+          and repost_item_id = track_id
+        order by created_at desc
+      ) as reposted_by,
+    
+      array(
+        select user_id 
+        from saves
+        where
+          is_current = true
+          and is_delete = false
+          and save_type = 'track' 
+          and save_item_id = track_id
+        order by created_at desc
+      ) as saved_by
+    
+    from tracks
+    join users on owner_id = user_id 
+    join aggregate_user on users.user_id = aggregate_user.user_id
+      WHERE tracks.is_current = true 
+        AND users.is_current = true
+    `
+
+    if (checkpoint.tracks > 0) {
+      // Find stale tracks including saves, reposts
+      sql += `
+      and track_id in (
+        select track_id from tracks where is_current and blocknumber >= ${checkpoint.tracks}
+        union
+        select save_item_id from saves where is_current and save_type = 'track' and blocknumber >= ${checkpoint.saves}
+        union
+        select repost_item_id from reposts where is_current and repost_type = 'track' and blocknumber >= ${checkpoint.reposts}
+      )`
+    }
+
+    return sql
+  },
+
+  forEach: (row: TrackDoc) => {
+    row.tags = row.tags ? row.tags.split(',') : []
+    row.repost_count = row.reposted_by.length
+    row.save_count = row.saved_by.length
+
+    row.length = Math.ceil(
+      row.track_segments.reduce((acc, s) => acc + parseFloat(s.duration), 0)
+    )
+
+    // some alternate routes...
+    // unshift because last one is_current
+    let slug = row.route_id.substring(row.route_id.indexOf('/') + 1)
+    row.routes.unshift(slug)
+    row.routes.unshift(`${slug}-${row.track_id}`)
```

Comment:
I think slug should always be enough. I don't think `${slug}-{id}` is valid b/c the slug includes the id for "legacy" tracks. right @rickyrombo ?

Reply:
Yeah `routes` actually comes from joining in the `track_routes` table.

I was just adding entries because of existing URLs like `/stereosteve/1k-453170`... 
but I guess there's client side logic to handle that?

I can remove the code that adds `route_id` and `${slug}-${row.track_id}` if we don't need these for track lookup... 

It seems to me there are at least 3 "route signatures" in circulation today, but we can also deal with this when actually implementing `get_track`

Reply:
Yeah, so there's n-many routes that resolve to a track, but each track always has exactly one that is the current route as returned by the API. I don't think we can guarantee the ${slug}-${id} will always be there. We should also probably be removing `route_id` from the table at some point. Probably best to leave this out? Probably better to rely solely on track_routes entries

Reply:
Yeah... so with elasticsearch indexing we should think about the "resolve" side of things... if there are `n-many routes that resolve to a track` we should try to index all `n` of them so that we can find a track without successive round trips.

The API can always choose the "last" one of `n` if we index it such that the last one is the `is_current`... but yeah that's the idea behind the speculative code I added and then removed.